corehq/apps/ota/utils.py

```diff
@@ -87,7 +90,50 @@ def demo_restore_date_created(commcare_user):
             return restore.timestamp_created
 
 
-def is_permitted_to_restore(domain, couch_user, as_user, has_data_cleanup_privelege):
+def _ensure_valid_domain(domain, couch_user):
+    if not couch_user.is_member_of(domain):
+        raise RestorePermissionDenied(u"{} was not in the domain {}".format(couch_user.username, domain))
+
+
+def _ensure_cleanup_permission(domain, couch_user, as_user, has_data_cleanup_privilege):
+    if not has_data_cleanup_privilege and not couch_user.is_superuser:
+        raise RestorePermissionDenied(u"{} does not have permissions to restore as {}".format(
+            couch_user.username,
+            as_user,
+        ))
+
+
+def _ensure_valid_restore_as_user(domain, couch_user, as_user):
+    try:
+        username = as_user.split('@')[0]
+        user_domain = as_user.split('@')[1]
+    except IndexError:
+        raise RestorePermissionDenied(
+            u"Invalid restore user {}. Format is <user>@<domain>".format(as_user)
```

Comment:
yeah i think worth it, though it's likely that no one will be able to do anything with these messages except us